### DISCUSSION TITLE
properties 파일 분리

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#187] [뜨거운 토론]
> to @rawfishthelgh @Kim0914 @yeonkkk 

## 주제

4차데모데이에서 production 서버 증설 혹은 환경분리를 생각하여 dev, prod 서버를 분리해야할 것 같다는 생각이 들었습니다.
이에 현재 설정파일을 덮어씌우는 현 방식에서 prod-application.properties, local-application.properties, dev-application.properties로 설정파일을 분리하고 각각의 값에 대해 서버의 환경변수로 구성하는게 어떨까? 라는 의견입니다.

환경변수 설정에는 두가지 방식이 존재할 것 같습니다.
1. production 서버의 OS 환경변수
2. build 서버 (젠킨스)의 환경변수로 관리하여 빌드할 때 값 넣어주기

properties 파일 분리에 대한 여러분들의 의견이 궁금하네요

**Comment by rawfishthelgh** [comment]
저는 2번 방식에 한 표를 던지겠습니다ㅎㅎ
프로퍼티 파일의 환경변수를 private repo를 활용해 서브모듈로 관리하고 빌드 시 넣어주는 방식을 활용하면,
ec2 인스턴스가 내려가더라도 환경변수를 유지할 수 있을 뿐 더러 
향후 빌드서버에 개발서버를 구성하는 방식을 택할 때 환경변수를 따로따로가 아닌 한 곳에서 관리할 수 있는 이점이 있을 것 같네요!

**Comment by Choi-JJunho** [comment]
갑자기 든 생각인데 private 레포지토리에서 설정파일을 관리하는 방법도 생각해볼 수 있을것 같네요

이 경우 submodule 기능을 이용해서 설정파일을 GitHub에서 끌어온 뒤 빌드 및 배포할 수 있다는 특징이 존재합니다.

**Comment by yeonkkk** [comment]
저도 submodule 을 이용하는게 좋을 것 같아용.
private repo 이용하면 깃허브를 이용하기 때문에  서버 내에서 관리하는 것 보다 조회와 수정이 편할 것 같습니다!
그리고 변경 히스토리 확인도 가능해서 유용할 것 같아요!
